File: packages/python/src/agent_wallet/core/utils/env.py
"""Shared env parsing helpers."""

from __future__ import annotations

from collections.abc import Mapping


def clean_env_value(env: Mapping[str, str], key: str) -> str | None:
    value = env.get(key)
    if value is None:
        return None
    trimmed = value.strip()
    return trimmed or None


def first_env(env: Mapping[str, str], keys: tuple[str, ...]) -> str | None:
    for key in keys:
        value = clean_env_value(env, key)
        if value:
            return value
    return None


def parse_account_index(value: str | None) -> int:
    if not value:
        return 0
    if not value.isdigit():
        raise ValueError("AGENT_WALLET_MNEMONIC_ACCOUNT_INDEX must be a non-negative integer")
    return int(value)

File: packages/typescript/src/core/utils/env.ts
export function cleanEnvValue(value: string | undefined): string | undefined {
  const trimmed = value?.trim()
  return trimmed || undefined
}

export function firstEnv(env: NodeJS.ProcessEnv, keys: readonly string[]): string | undefined {
  for (const key of keys) {
    const value = cleanEnvValue(env[key])
    if (value !== undefined) return value
  }
  return undefined
}

export function parseAccountIndex(value: string | undefined): number {
  const normalized = value?.trim()
  if (!normalized) return 0
  if (!/^\d+$/.test(normalized)) {
    throw new Error('AGENT_WALLET_MNEMONIC_ACCOUNT_INDEX must be a non-negative integer')
  }
  return Number(normalized)
}
